### PR TITLE
adjust to the Podfile generate_multiple_pod_projects=true mode

### PR DESCRIPTION
**Pull request by wangzhizhou**
try to fix issue #130

### CODE REVIEW

File: lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb [hunk 1]
```diff
@@ -8,13 +8,32 @@ class AddSpmPkgs < Hook
         def run
           return if @spm_resolver.result.spm_pkgs.empty?
 
-          add_spm_pkg_refs_to_project
-          add_spm_products_to_targets
-          pods_project.save
+          if podfile.installation_options.generate_multiple_pod_projects
```
Comment (trinhngocthuyen) [inline lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb:11]: to simplify the logic, how about this way (not having if/else check):
```ruby
projects_to_integrate.each do |project|
  add_spm_pkg_refs_to_project(project)
  add_spm_products_to_targets(project)
  project.save
end
```
where `projects_to_integrate` is:
```ruby
# in base.rb
def projects_to_integrate
  [pods_project] + @context.pod_target_subprojects
end
```

File: examples/Podfile [hunk 2]
```diff
@@ -26,14 +30,21 @@ target "EX" do
...
+    pod "Orcam", :macro => {
+      :git => "https://github.com/trinhngocthuyen/orcam.git",
+      :branch => "main",
+    }
+    pod "MacroCodableKit", :macro => {
+      :git => "https://github.com/mikhailmaslo/macro-codable-kit",
+      :tag => "0.3.0",
+    }
+  else
+    spm_pkg "Orcam", :git => "https://github.com/trinhngocthuyen/orcam.git", :branch => "main"
```
Comment (trinhngocthuyen) [inline examples/Podfile:45]: I think macro package is just like any other package which can be integrated with `spm_pkg`. So we don't necessarily need to have if/else here :)
If we do wanna showcase that macro can be integrated in both ways, we can consider having one of them with `pod ..., :macro => ...` and the other one with `spm_pkg`.
```ruby
pod "Orcam", :macro => {
  :git => "https://github.com/trinhngocthuyen/orcam.git",
  :branch => "main",
}

spm_pkg "Orcam", :git => "https://github.com/trinhngocthuyen/orcam.git", :branch => "main"
```

Comment (trinhngocthuyen) [reply examples/Podfile:45]: The reason I integrated both of them with `pod ..., :macro => ...` is to reduce the integration test time (on Github Actions).
It takes quite some time to build swift-syntax packages LOL.

File: lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb (hunk 1 above)
Comment (wangzhizhou) [reply lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb]: it's great, i will have a try.

File: examples/Podfile (hunk 2 above)
Comment (wangzhizhou) [reply examples/Podfile:45]: I will revert these changes to take the prebuild action

File: lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb (hunk 1 above)
Comment (wangzhizhou) [reply lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb]: pods_project should be saved last, or project info could be write into disk

Comment (wangzhizhou) [reply lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb]: after execute as suggestion above, every pod_target_subproject will dep all spms in there root_object and targets, which may not be correct

Comment (wangzhizhou) [reply lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb]: today was will try to refine the code as you suggest, maybe do some Chang

Comment (wangzhizhou) [reply lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb]: I refined the spms add logic, add spm_ref to targets of project first, and then add the  pgk_refs to the project.

**Comment by wangzhizhou** [comment]
Does the pod 'xxx', :macro => { ... } support :path specify the local swift macro package
image

**Comment by wangzhizhou** [comment]
and prebuild not work as description, may be swift compiler warning as follow is the reason:
image

the prebuild product will be added the '-tool' suffix, which can't be found when the plugin copy it 
image

image

**Pull request closed without merging**